### PR TITLE
Some updates; mostly issues with the ECS agent re-registering itself on reboot.

### PR DESCRIPTION
**Pull request by gswallow**
See http://docs.aws.amazon.com/AmazonECS/latest/developerguide/ECS_instances.html#container_instance_life_cycle for their description of a bug in the ECS agent < 1.8.1.

### CODE REVIEW

File: ansible/roles/docker/tasks/main.yml [hunk 1]
```diff
@@ -42,7 +42,7 @@
 - apt: name=apparmor
   tags: 
     - build_ami
-- apt: name=docker-engine=1.7.0-0~trusty
+- apt: name=docker-engine=1.9.1-0~trusty
```
Comment (phobologic) [inline ansible/roles/docker/tasks/main.yml:61]: We've ran into issues w/ 1.9 - honestly anything past 1.7 has led to issues with the logging infrastructure.  We're waiting for 1.11, which willhopefully fix some of those issues with the new logging drivers, and then we'll be releasing a new logging infrastructure along with it.

Comment (gswallow) [inline ansible/roles/docker/tasks/main.yml:61]: Docker 1.11 is out afaik.  I'm wondering if the round-robin DNS features in Docker 1.11 mean that ECS will schedule more than one web worker on a minion?

I figured that my update to docker engine 1.9.1 might raise questions.  I think we got rid of the cloudwatch/kinesis logging in lieu of shipping logs directly to sumologic.

Comment (phobologic) [inline ansible/roles/docker/tasks/main.yml:61]: Yep - just got released - we're hoping to get a chance to look at it soon.  Eric did a bunch of work testing out 1.10 with the new logging drivers, but the performance was poor (and the API logging performance fell apart as well - https://github.com/docker/docker/issues/20586).  I can't actually remember what issues we ran into with 1.9.

**Comment by phobologic** [comment]
Hey @gswallow - so we're actually in the works of releasing a new version of the AMI & the stacker blueprints.  @mhahn is leading that effort, and it's going to include a _lot_ of new stuff, including docker 1.11.2, the latest ECS, etc.  You can find those efforts here: https://github.com/remind101/empire_ami/pull/26

Not sure if we should close this and potentially roll your changes up into the AMI that we're working on releasing, or what.  I actually don't have a strong handle, yet, on what the direct-lvm mode change will mean.

**Comment by gswallow** [comment]
This particular issue, I think it's safe to close.  I was talking to you guys in the slack room today and I was asked by @ejholmes to submit an issue.

Since we are already using direct-lvm mode, and I had the changes to Ansible, I figured I would just submit a pull request.

**Comment by phobologic** [comment]
Cool - thanks!

File: ansible/roles/aws_ecs/tasks/main.yml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 - include_vars: aws_ecs.yml
-- file: name=/var/run/ecs_agent mode=0750 owner=root group=root state=directory
+- file: name=/var/lib/ecs_agent mode=0750 owner=root group=root state=directory
```
Comment (gswallow) [inline ansible/roles/aws_ecs/tasks/main.yml:2]: This change should be considered.  /var/run gets wiped out on reboot, and the ECS agent registers itself as a new instance every time a minion is restarted.

**Comment by gswallow** [comment]
Oh, damn.

There _is_ a change in there you should consider.  I added a comment to the
line that covers it.

File: ansible/roles/aws_ecs/tasks/main.yml (hunk 2 above)
Comment (mwildehahn) [inline ansible/roles/aws_ecs/tasks/main.yml:2]: verified this is happening

Comment (mwildehahn) [inline ansible/roles/aws_ecs/tasks/main.yml:2]: will include in the `empire-daemon` PR

**Pull request closed without merging**